### ISSUE TITLE
Lazy-Loading in Joomla core

### ISSUE THREAD

**Issue by 4digitalworld** [joomla/joomla-cms#27761]
I suggest turning on lazy-loading for the **Control Panel** and **Frontend**.

**Why native lazy-loading?**
According to [HTTPArchive](https://httparchive.org/reports/page-weight), images are the most requested asset type for most websites and usually take up more bandwidth than any other resource. At the 90th percentile, sites send about 4.7 MB of images on desktop and mobile. That's a lot of [cat pictures](https://en.wikipedia.org/wiki/Cats_and_the_Internet).

Embedded iframes also use a lot of data and can harm page performance. Only loading non-critical, below-the-fold images and iframes when the user is likely to see them improves page load times, minimizes user bandwidth, and reduces memory usage.

Currently, there are two ways to defer the loading of off-screen images and iframes:

- Using the [Intersection Observer API](https://developers.google.com/web/updates/2016/04/intersectionobserver)
- Using `scroll`, `resize`, or `orientationchange` [event handlers](https://developers.google.com/web/fundamentals/performance/lazy-loading-guidance/images-and-video/#using_event_handlers_the_most_compatible_way)

Either option can let developers include lazy-loading functionality, and many developers have built third-party libraries to provide abstractions that are even easier to use. With lazy-loading supported directly by the browser, however, there's no need for an external library. Native lazy loading also ensures that deferred loading of images and iframes still works even if JavaScript is disabled on the client.

Support for natively lazy-loading images and iframes is coming to the web! This video shows a [demo](https://mathiasbynens.be/demo/img-loading-lazy) of the feature: [https://web.dev/native-lazy-loading/lazyload.webm](https://web.dev/native-lazy-loading/lazyload.webm)

**The loading attribute**
Here are the supported values for the `loading` attribute:

- `auto:` Default lazy-loading behavior of the browser, which is the same as not including the attribute.
- `lazy:` Defer loading of the resource until it reaches a calculated distance from the viewport.
- `eager:` Load the resource immediately, regardless of where it's located on the page.

**Example:**
`<img src="image.png" loading="lazy" alt="…" width="200" height="200">`
`<iframe src="https://example.com" loading="lazy"></iframe>`

**Media Manager issue:** [https://github.com/joomla/joomla-cms/issues/21936](https://github.com/joomla/joomla-cms/issues/21936)
**Source:** [https://web.dev/native-lazy-loading/](https://web.dev/native-lazy-loading/)
**Lazy-Loading Images in WordPress Core:** [https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/)
**Wiki:** [https://en.wikipedia.org/wiki/Lazy_loading](https://en.wikipedia.org/wiki/Lazy_loading)

**Comment by brianteeman** [comment]
https://caniuse.com/#feat=loading-lazy-attr

**Comment by 4digitalworld** [comment]
> 
> 
> https://caniuse.com/#feat=loading-lazy-attr

This will not break websites if we add this attribute. Nothing will happen if the user logs in from an old browser to a website.

But it will help everyone speed up the website, who use the new version of the browser.

This does not prevent the most popular CMS Wordpress from including **lazy-loading** support in the next version. Why should Joomla default by technology worse, than Wordpress??

[https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/)

**Comment by 4digitalworld** [comment]
@brianteeman One way or another, it can be turned on for Joomla 4 by default, and for Joomla 3 to turn off this parameter, with the ability to turn it on.

**Comment by 4digitalworld** [comment]
@joelsan I have the same problem and here is its solution!

**Comment by brianteeman** [comment]
Sorry for the quick reply before. The explanation behind making that reference is that it has always been a policy of Joomla that any core feature must be available on all supported platforms and browsers.

You are correct that it wont break a website but it will most likely confuse users for whom a feature does not work.

**Comment by ReLater** [comment]
> This does not prevent the most popular CMS Wordpress from including lazy-loading support in the next version.

I can't find any evidence on the page you've linked to that is proving this statement. i only see a discussion there. And I find some good arguments there to NOT integrate it in core in a overhasty way.

**Comment by 4digitalworld** [comment]
> You are correct that it wont break a website 

I believe that if it is in demand and it does not break the website, then it can be done! One way or another, in 1-2 years it will be available to more than 95% of users, so why not do it now, in advance? But we want to develop the global web, and most importantly - it does not break the website.

What difference does it add in 2 years, when it will be available to everyone, or now, if it does not break the website?

- [x] Most importantly: **THIS DOES NOT BREAK THE WEBSITE**.

**Comment by mbabker** [comment]
> What difference does it add in 2 years, when it will be available to everyone, or now, if it does not break the website?

Because the general philosophy in maintaining Joomla is that it does not conditionally introduce features that are only available to users running on the most current PHP platforms, or web server platforms, or visitors using the latest browsers; the core feature set is in part decided upon by what is supported in the lowest versions of whatever Joomla claims to support, or if a feature can be polyfilled from newer versions of something to be used in older versions.

It's not a matter of "well it doesn't break the website so it's OK".

And, while it might be convenient if Joomla's WYSIWYG integrations can add support for the lazy loading attribute, you can add it in the HTML code of your existing content today; you don't need to wait for Joomla to expose it in the editors somehow.

**Comment by 4digitalworld** [comment]
@mbabker I think that we need to change the philosophy of Joomla. And highlight the most important: 

- [x] 1) This is in demand
- [x] 2) This does not break the website

And if both points are fulfilled, then you can add this to the CMS. But there is nothing wrong if the CMS will be developed? Why are new features added late in 1-2 years? You can do this now and wait for others to update their browsers. This does not break any website, so it can be added, right?

I remember that I suggested some important things for the CMS, and they were added only after a few months or years. Why wait all the time when it can be used right now? Other CMS are adding something new, why should Joomla be worse off?

Why Wordpress and Drupal do not have problems with adding new technologies, and only Joomla does not want to add something new. But this does not break existing websites, why then cannot it be added?

**I think that we need to change the philosophy for the CMS and be open to new technologies if these technologies do not break existing websites.**

**Comment by mbabker** [comment]
Nobody said Joomla is not open to new technologies.  What people are saying is that Joomla is not built in a way that it delivers a different experience based on if the website is running on a server powered by PHP 5.3 or 7.3, or if the application is interacted with by someone using Internet Explorer 11 or Edge 18.  It doesn't matter if "using this new feature doesn't break a website".

Your evaluation criteria clearly screams out "I am an end user and just want this now".  Your evaluation criteria is lacking other critical technological factors that need to be considered in the decision making process in for a mass distributed platform.

And again, Joomla does not need to add any form of support for the lazy loading attribute for you to use it.  Seriously.  Hell, you could go back to a Joomla 1.0 website, load up the article editor, and add the attribute in the image's HTML tag.  The only thing this feature request sanely does would be to add a shortcut to the WYSIWYG editor (assuming it's available in the included version of TinyMCE) and possibly the media field, which are used to auto-generate the HTML tag inserted into something.

**Comment by jeremysmartin399** [comment]
@ReLater If I'm reading the [WP article about adding lazy loading into WP core](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/) correctly, they plan to add lazy-loading to WP 5.4 in the next 2 weeks. Some commented that 2 weeks felt rushed, but I could not find any comments from developers with concerns about users with different browsers having a different experience. (eager loading as opposed to lazy loading)

FWIW, according to caniuse, the [loading-lazy-attr is supported by browsers used by 62.19% of worldwide audience](https://caniuse.com/#feat=loading-lazy-attr).

**Comment by gerryfrancis** [comment]
I understand both reasonings. If someone wants to have it, probably some overrides will do.

I would not like to see this as default behavior that cannot be changed. When implemented, admins must have the possibility to switch it on or off.

**Comment by 4digitalworld** [comment]
> 
> 
> When implemented, admins must have the possibility to switch it on or off.

Absolutely right! I suggest making such a setting in Joomla. I think it will be useful to very, very many administrators and users of websites.

I have one problem. I'm upload over one thousand images in one folder, and when I open the media manager, a lot of RAM is taken from my computer. I think so many website administrators have a similar problem with thousands of articles on their websites.

**Comment by brianteeman** [comment]
That is a completely different issue and would have been resolved by you if you had not put all your images in the same folder. Your operating system has the exact same issue.

I am closing this issue at this time. If and when lazy-loading is available natively in all supported browsers it can be looked at again.

For now you will find many extensions available that will do this for you n your site.

**Comment by 4digitalworld** [comment]
@brianteeman I'm not even surprised why there are so few developers left who are still interested in developing Joomla. Any normal idea is rejected and Joomla remains the same as it was in 2005, only the design of the control panel changes. That’s all I’ve seen in the last 15 years. Only a change of design for the control panel.

I see so many opportunities in other CMS, such as WordPress and Drupal and I do not see anything of this in Joomla. Other CMSs also have additional extensions, but other CMSs are developing as a convenient tool for website editors. Joomla has not developed in this way since 2005. The world has changed in 15 years, the Internet has changed in 15 years, and you still think the same way as in 2005.

**Comment by jeremysmartin399** [comment]
@sanek4life  Maybe [LLFJ - Lazy Load for Joomla! ](https://extensions.joomla.org/extension/lazy-load-for-joomla/)would work for your sites? It is free with a 5 star rating on JED with 42 reviews. (It does use Mootools)  Or maybe a plugin could be developed to add lazy loading on Joomla! 4? With [Wordpress appearing to add lazy loading to core](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/), I think eventually other browsers will support the loading="lazy". Then maybe it could be added to Joomla 4.x.

**Comment by 4digitalworld** [comment]
> 
> 
> maybe a plugin could be developed to add lazy loading on Joomla! 4? With [Wordpress appearing to add lazy loading to core](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/), I think eventually other browsers will support the loading="lazy". Then maybe it could be added to Joomla 4.x.

I think that the leadership of Joomla needs to decide that there are some things that should be **native** for Joomla, and not as extensions from third-party developers. At the very least, one could make official extensions for some ideas, as was done with the **[Weblinks](https://extensions.joomla.org/extension/official-extensions/weblinks/)**.

After migrating from 1.5 to 2.5, many good extensions were no longer supported. I saw the same thing when I migrated from 2.5 to 3.x, and now I see that after 3.x some extensions will not work on 4.x

Third-party developers come and go, and after that each administrator and editor of the website has many problems with the migration of his website.

I don’t see a problem if the slider or any modul is no longer supported - such things can be replaced with others in the [JED](https://extensions.joomla.org/) extensions directory. But I believe that the basic things for editing articles should be at the core of Joomla, or at least be presented as **Official Extensions** here - [https://extensions.joomla.org/category/official-extensions/](https://extensions.joomla.org/category/official-extensions/)

I think that on the verge of a crisis to transition to Joomla 4, the leadership of Joomla 4 should reconsider its attitude towards website editors and the entire development strategy of the CMS. I did not offer anything bad for the CMS. I offered really good tools for editors. I do not think that these things should be from third-party developers. Third-party developers can make components for the online store, forums, widgets in the form of modules and other things that do not affect the editing of articles in Joomla. Everything related to `com_content` should be native if it does not break existing websites. I really don’t understand why I meet so much resistance.

**Comment by mbabker** [comment]
> I really don’t understand why I meet so much resistance.

Then you really do not understand the amount of effort it takes to maintain a general purpose content management system designed for a multitude of use cases by a multitude of individuals and organizations, and inherently why said content management system becomes unmaintainable if everything is included in core.  Take a step back and stop looking at things as an entitled consumer of the software and understand that there are a vast number of decisions that must be made regarding the software and what features are acceptable for core versus extensions and when it is appropriate to introduce new features into core.  Until such a time that you understand such methodologies and that Joomla is not a "GIVE ME WHAT I WANT" piece of software you can make demands of, discussing any feature proposal with you is a futile effort.

**Comment by ssnobben** [comment]
Its understandable to have a dev strategy and formalised guidelines what kind of general core features Joomla should have and thanks for clearing that up. 

Is there any policy document regarding what requirements and functions decide these feature conditions?

And that mean that the effort to put something that is very hard to maintain ie Pagebuilder will and shall not be included in Joomla core bcs its a nightmare to maintain and there are already many "Pagebuilders" 3pds that are really good and working well for Joomla.

**Comment by 4digitalworld** [comment]
@mbabker Can it be used in Joomla (сan this code be ported to the CMS?)? Here is the plugin for WordPress: [https://github.com/WordPress/wp-lazy-loading](https://github.com/WordPress/wp-lazy-loading)

**Comment by gerryfrancis** [comment]
@sanek4life I do not know if you ever heard of plugins for Joomla!, but I have found "LLFJ - Lazy Load for Joomla!". I have no experience with that, maybe you give it a try.

> the plugin for WordPress

See? The WordPress core does not include every feature you can think of as well...

**Comment by ReLater** [comment]
@sanek4life 

- It's not too difficult to build a Joomla plugin that sets `loading="lazy"` to any image on-the-fly while the page HTML is rendered.

- But it's not easy to avoid performance issues or other issues if you're running JavaScript based plugins in parallel like "LLFJ - Lazy Load for Joomla!" written by @Kubik-Rubik THAT FUNCTIONS WITH ALL BROWSERS.

- The goal for Joomla **core** is that **ALL** supported browsers support lazy image loading if you provide a feature like that. It's a No-no to ignore that promise just to have this or that feature in core! Only for some users...

- Only browsers that understand `loading="lazy"` should get this attribute for images. **But most important:** Only browsers that don't understand the attribute should fall back to the JS solution BUT NOT BROWSERS THAT UNDERSTAND THE ATTRIBUTE. The latter ones should not load the JavaScript unnecessarily.

Means: A plugin or core or whoever or whatever must detect the browser to make this descision. Otherwise some browsers will run through BOTH solutions. The performance win on the one hand is a performance loss on the other.

The browser detection is an additional performance loss. I don't know how grave it is or if there exist modern solutions to detect if the current browser supports this or that.

**Comment by 4digitalworld** [comment]
> 
> It's not too difficult to build a Joomla plugin that sets `loading="lazy"` to any image on-the-fly while the page HTML is rendered.

I am not a developer, **if you do, it will be great**. That’s all I wanted to have such a plugin for both the **frontend** and the **backend** - `Control Panel` - `Media Manager` (I and many other editors really need this plugin. I wrote more than 5000 articles, I probably understand better what is really needed for the **CMS = content management system**. I don’t do websites, of which there are 5 categories and 30 articles, I make websites that have 300 categories, 600 tags and 5000 articles. Do you understand this? Joomla is not for everyone and not for everyone in this case.)

Perhaps I'm generally the only editor (not a developer) who communicates with you here on a GitHub.

> 
> But it's not easy to avoid performance issues or other issues if you're running JavaScript based plugins in parallel like "LLFJ - Lazy Load for Joomla!" written by @Kubik-Rubik THAT FUNCTIONS WITH ALL BROWSERS.

This plugin is not suitable for me because it does not work natively in the browser. This plugin uses javascript. This is not a suitable solution. I want everyone who has a modern browser to have `Lazy Loading` benefits. People do not view my site with an `Internet Explorer`.

> 
> The goal for Joomla **core** is that **ALL** supported browsers support lazy image loading if you provide a feature like that. It's a No-no to ignore that promise just to have this or that feature in core! Only for some users...

Why does this not prevent the more popular CMS - @Wordpress from including this in the core? What is the problem with @Joomla? Is this CMS not for everyone?

![изображение](https://user-images.githubusercontent.com/3662519/74641398-45fc9180-5182-11ea-9a1b-09b54b497d8f.png)

**Source:** [https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/)
> 
> Only browsers that understand `loading="lazy"` should get this attribute for images. **But most important:** Only browsers that don't understand the attribute should fall back to the JS solution BUT NOT BROWSERS THAT UNDERSTAND THE ATTRIBUTE. The latter ones should not load the JavaScript unnecessarily.

I don't care what people who do not update their browser do. They are not the target audience of my site. And they will not have problems because of `Lazy Loading`, their browser will ignore this parameter and they will not have problems. If they update their browser, then they will see the benefits. `Lazy Loading` will not break their old browser or break existing sites on Joomla. I do not understand what the problem is?

> 
> 
> Means: A plugin or core or whoever or whatever must detect the browser to make this descision. Otherwise some browsers will run through BOTH solutions. The performance win on the one hand is a performance loss on the other.

Why are you writing about this? I do not want to use the javascript plugin, I only want to use parameter `Lazy Loading`. If a person uses an old browser, it is his problem that the site will load longer. He can always update his browser and then he will see that the site loads faster.

> 
> The browser detection is an additional performance loss. I don't know how grave it is or if there exist modern solutions to detect if the current browser supports this or that.

This is not needed, and javascript is not needed. Why are you trying to make life more complicated than it is?

**Comment by 4digitalworld** [comment]
@gerryfrancis @ReLater read what is written **[here](https://make.wordpress.org/core/2020/01/29/lazy-loading-images-in-wordpress-core/)**! @Wordpress does not offer to use javascript, they just offer to use `Lazy Loading` natively in the browser, without downloading scripts! Why are you writing about performance? Have you ever read the original publication from Woprdress?

**Comment by brianteeman** [comment]
You got it exactly right when you said joomla is for everyone - thats why this cant be in the core.

Updating a browser is not always possible. Especially if you are in a corporate world or dont have the latest phone.

Thats why joomla is customisable to give you the opportunity to not care about people and install an extension.

Joomla can not nor should it ever contain everything that everyone wants.

At the end of the day if you want it so much for your web site you will find a way to achieve it.

**Comment by 4digitalworld** [comment]
> 
> 
> You got it exactly right when you said joomla is for everyone - thats why this cant be in the core.
> 
> Updating a browser is not always possible. Especially if you are in a corporate world or dont have the latest phone.
> 
> Thats why joomla is customisable to give you the opportunity to not care about people and install an extension.
> 
> Joomla can not nor should it ever contain everything that everyone wants.
> 
> At the end of the day if you want it so much for your web site you will find a way to achieve it.

These plugins for Joomla use javascript, they do not use it natively in the browser.

If this does not break old browsers, why can't it be added to the CMS? This is not a script, it's just a **[tag for images](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-loading)**. This will not break old browsers.

```
<img src="example.img" loading="lazy" alt="example1" width="200" height="200">
<img src="example.img" loading="lazy" alt="example2" style="height:200px; width:200px;">
<!-- lazy-loaded -->
```

I understand when new scripts are added and it breaks old browsers, but in this case it does not break anything. This can be done as an official plugin for Joomla, which can be turned `on` or `off`, like other plugins (for example, plugins - `Tags` or `Fields`). And in the plugin you can specify whether to use it only for `Control Panel` or for the `Frontend` (or in both cases), or disable the plugin. Why is this such a big problem?

Currently, there is not a single third-party extension for Joomla that natively uses `Lazy Loading` in the browser, so I wrote about it here on the GitHub. And also I saw that @Wordpress includes Lazy Loading in its CMS and they do not have any problems with it.

> According to HTTPArchive, images are the most requested asset type for most websites and usually take up more bandwidth than any other resource. At the 90th percentile, sites send about 4.7 MB of images on desktop and mobile.
>     [Native lazy-loading for the web, web.dev](https://web.dev/native-lazy-loading/) 

I can confirm that this problem exists. There are pages on my website where 100 `articles` are displayed in the category (this is all on one page) and each article has an `intro image` that weighs about 100 kb. I think you understand that I am not the only one on this Internet who publishes images on the website. Another user has a similar problem - [https://github.com/joomla/joomla-cms/issues/21936](https://github.com/joomla/joomla-cms/issues/21936)

This can be solved natively through a browser, without using javascript. This plugin will be included in the new version of WordPress - [https://wordpress.org/plugins/wp-lazy-loading/](https://wordpress.org/plugins/wp-lazy-loading/)

**Comment by ssnobben** [comment]
Its Joomla problem dilemma and would be good to see the analysis of why WP got to do this decision technically and why its not good for solution for Joomla right now or in future. 

Its difficult decisions but if WP can maybe its possible for Joomla too? or we miss the large blog groups site owners to swing over to use 100 % Wordpress instead ( Joomla for all.. but not for large blogger system..)  but as said:

**You got it exactly right when you said _joomla is for everyone_ - thats why this cant be in the core.** hence not large blogger groups its better they use Wordpress bcs its not Joomla focus ie for everyone.

And then opposite..

**_Joomla can not nor should it ever contain everything that everyone wants._**

Sometimes you have to take decisions against that 3 persons on the planet or other relevant data that stop Joomla from evolving from Dinosaur times. 

I just wonder who that important “Joomla for everyone” is?  I never see him, heard of him or met him but he seems to be very special :)

**Comment by gerryfrancis** [comment]
@sanek4life 

> I do not understand what the problem is?

You are so right.

I guess implementing Lazy-Loading in Joomla! core is not a problem. Sure, it requires the skills of some developers, but I am convinced they can make it. And administrating/supporting this function in the long run would be possible, too.

The problem is that you are a person who is demanding something that, in YOUR eyes, people MUST supply for YOUR use case. And you are "selling" YOUR demand as it was the wish of all Joomla! users. ("We wants it, we NEEDS it, yes, YESSS!!" NO, because you cannot speak for everyone!) You are wasting lots and lots of your energy in complaining and demanding. How many times have we told you that, for YOUR use case, you can achieve things like that with an override? If you were asking/searching (in a Joomla! forum) how to write YOUR override to get YOUR solution for YOUR site - and I am sure someone would see this possibility as a welcomed challenge and do this favour to you, maybe even FOR FREE -, you would not argue here any longer. So get your head out of your b*tt, stop crying, and help yourself, will you? And maybe even try to accept that neither developers nor volunteers are bond servants.

@ssnobben

> I just wonder who that important “Joomla for everyone” is? I never see him, heard of him or met him but he seems to be very special :)

Made my day, thanks! :)

**Comment by 4digitalworld** [comment]
> 
> You are so right.
> 
> I guess implementing Lazy-Loading in Joomla! core is not a problem. Sure, it requires the skills of some developers, but I am convinced they can make it. And administrating/supporting this function in the long run would be possible, too.

**If it’s not difficult**, why are so many people here against and no one even said that they can offer their code? They just wrote to me that it is very difficult to support and no one wants and will not do it.

> 
> maybe even FOR FREE

You are very focused on money. Why are you using `open source` then? There are many paid solutions. I don’t take money from readers who visit my website. Imagine that people used to buy magazines and pay money to read something. The world has changed, now almost everything is `open source`, this is not a topic for discussion here.

**Comment by 4digitalworld** [comment]
> 
> Sometimes you have to take decisions against that 3 persons on the planet or other relevant data that stop Joomla from evolving from Dinosaur times.
> 
> I just wonder who that important “Joomla for everyone” is? I never see him, heard of him or met him but he seems to be very special :)

It seems to me that the general position of key people in @Joomla is to use `Windows XP` and `Internet Explorer 6`. I cannot explain this in any other way.

These people, whose browser does not support new technologies, most likely still use `Windows XP`. Even @Microsoft says - support for `Windows 7` **[is over](https://support.microsoft.com/en-us/help/4467761/windows-what-happens-when-windows-7-support-ends)**, use only `Windows 10`.

Who are these people who do not use modern browsers? Probably, these are the very 1-2% that are still using `Windows XP`, at the same time that Microsoft is telling the whole world that the minimum requirements for today are `Windows 10`.

The creators of @Wordpress have a clear mind and use the brain correctly, so using them `Lazy Loading` is not a problem.

I know that there are corporations that use old browsers - these corporations also use old programs. Why do not they continue to use `Joomla 3` (or even `Joomla 2.5`???)? This function could appear in `Joomla 4`. This function could appear in Joomla 3 as a `plugin` that would be disabled by default. I think that the site administrator on Joomla, who works in the corporate world understands whether to use this function or not.

`Joomla 4` should not support `Windows 7`, and the necessary minimum should be - `Windows 10` and the Edge browser based on `Chromium`.

**If they like old technologies, they can use `Joomla 3.10`, which will be supported for 2 more years after the release. I think this has logic and that will be reasonable.**

**Comment by mbabker** [comment]
@sanek4life It is crystal clear you lack respect for users of software that do not agree with you 100% and do not use software in exactly the ways you do and at this point any discussion with you is a flat out waste of time.

**Comment by 4digitalworld** [comment]
> 
> 
> @sanek4life It is crystal clear you lack respect for users of software that do not agree with you 100% and do not use software in exactly the ways you do and at this point any discussion with you is a flat out waste of time.

It is obvious that you and people like you lack respect for editors who write articles on the Internet and publish anything.

Due to the fact that @Joomla is not very perfect, one day I had to go into several thousand articles on the website and replace the `flash` code from various audio and video players with `html5` code (without using flash technology), because browsers said that they will no longer support flash technology and by default this technology will be disabled!!!

What do you even know about the complexity of website support ?? You and people like you are very disrespectful to the website editors.

Why did such a situation occur when I had to manually redo the articles? Because in Joomla there is no built-in `shortcode` for displaying a player from `YouTube` or `SoundCloud` (in `WYSIWYG editor`).

And the second situation is when my website switched from `http` to `https`, and I manually had 2 thousand `articles` in which images were added along the absolute path (`http://site.com/image.jpg`), instead of the relative one (`/image.jpg`). This happened because the `WYSIWYG editor` from the **third-party developer** that I used automatically indicated the images in the absolute path (`http://site.com/image.jpg`) when I uploaded the images to the website. Later, the creators of the `WYSIWYG editor` solved this problem, but the old articles still showed images with an absolute path, so I had to manually change the image addresses without using the protocol (`http`) and site name (`site.com`).

Since then, I hate everything that needs to be done manually. Because when the CMS is not well developed, such situations arise. **And even third-party extensions do not helping** (which I wrote about in the situation with images).

**You do not respect EDITORS**. But you probably go to different websites and read various articles, right ??? So why don't you respect my work and the work of other editors??

When you say that a function is not needed, you are completely disrespectful to website editors! I write articles, I look at statistics, I see how quickly or slowly the website loads, I look at the positions of the website in the search, I write FOR PEOPLE. I see when it’s not convenient for people to use the website.

You do not fully respect the people who read articles on websites. What I am writing about is what editors need, because only editors understand these problems. You ignore editors' concerns, so you show your total disrespect for the work of editors.

Imagine that you get disrespect from yourself by the doctors or the seller in the store, only for the reason that you are not an expert in this aspect of life. They will tell you - study independently and do not ask for help.

You feel that you are GOD, but you are not a god, you are just the same person as me. You must explain to me humanly why I do not need this function and many other editors do not need  if you find at least one logical explanation for this, why this function is superfluous.

I do not ask to develop artificial intelligence or something super complex. I see only one negative message. This is a total disrespect for the person who writes the articles and is the editor.

**I wouldn’t write all this if WordPress didn’t make a very big announcement that they would introduce this function into the CMS. It turns out that the creators of @Wordpress (the more popular CMS) for you are some kind of fools who do not understand what they are doing. I'm right?**

**Comment by gerryfrancis** [comment]
@sanek4life 

> If it’s not difficult, why are so many people here against and no one even said that they can offer their code?

I never wrote that it was not difficult. When focusing on maintaining Joomla! in the long run, Lazy-Loading is another part that has to work together with so many others. You literally are calling people names why they do not support a function you are demanding, but you clearly have no clue what it takes to "make everything work with everything". You are just complaining, that seems to be all you can do - no, I am correcting myself: all you WANT to do!

> You are very focused on money.

Well, I drive an 11 year old car. Many thanks for the freshening laugh! :)

> Why are you using open source then? There are many paid solutions.

"Open Source" is no general term for software that can be used for free, but there are so many people like you who do not know this fact.

> It seems to me that the general position of key people in @joomla is to use Windows XP and Internet Explorer 6. I cannot explain this in any other way.

Because of "Open Source", software like Joomla! can be built and maintained by so many different people who take responsibility that (existing) things work, now and tomorrow, while you do not give a damn about users with older web browsers. Imagine a WWW where plain old HTML code cannot be loaded by current browsers. Have you ever developed solutions for corporate environments? No, you have not.

> I don’t take money from readers who visit my website. Imagine that people used to buy magazines and pay money to read something.

Don't worry, you will never get any money from me... ;)

> The creators of @WordPress have a clear mind and use the brain correctly, so using them Lazy Loading is not a problem.

Ah, so the Joomla! folks "do not use their brain correctly", is that what you wanted to tell? Sorry, that reminds me of lunatic people who claim everything was alright with themselves, but everybody else shall be mentally disturbed instead.

Do you know of the blind man who tells about the sun? The difference is: You are no blind man, you just do not want to see to understand. You are not even "lazy loading", you are just lazy...

And this is my last response, too. Be happy with any other CMS or not, but most of all TRY to become a better person who respects decisions made for the vast majority of so many people in the world. Unless you learn this important lesson in life, any further discussion with you really is a waste of precious time, even yours.

Good luck, all the best, and good-bye.

**Comment by ReLater** [comment]
Puuh! What a stupid discussion.

> I want everyone who has a modern browser to have Lazy Loading benefits. People do not view my site with an Internet Explorer.

"who has a modern browser"

Have a look on
https://caniuse.com/#feat=loading-lazy-attr

The red labeled browser versions DO NOT(!!!!) support this feature. For these browsers you need a fall back to JS. Even in Joomla 4.

**Comment by ssnobben** [comment]
This discussion reminds you about the early Gopher days before I had a chance to get the Netscape v 0.98 release and all the browser problem discussion with customers and even worse when IE coming onboard. My experience is that you can not be everything for everyone so why is a small tiny org like Joomla keen on being in that position that is very hard and difficult to keep? 

Market conditions and users expectations is different for Joomla now and Joomla today is another more complex larger system hard to maintain than 2005 fork and first attempt to be something on its own. So what control the decision strategy to prioritize what in core or not? I asked this question numerous times and nobody can answer this question bcs nobody knows.

**Comment by mbabker** [comment]
> It is obvious that you and people like you lack respect for editors who write articles on the Internet and publish anything.

I don't need to have my ego stroked, but since you seem to know nothing about me and my contributions to the open source community, allow me to stroke my ego for you.

In addition to having been a release manager for Joomla for a cumulative 4 years, personally being responsible for oversight and delivery of ~33% of all of Joomla's releases to its userbase, I am also the lead developer of a web agency which oversees the development and release of websites to over 50 clients built using a plethora of PHP and JavaScript based technologies (including Joomla) that satisfy requirements both for legacy platforms and modern technology stacks.

So, contrary to the crap that you want to publicly state because I will not become a slave to your demands, I am VERY aware of the needs of content editors and publishers, whether their needs are to be able to publish a simple blog or advanced concepts of content (such as auctioning platforms or ecommerce shops).

And, unlike you and your entitled attitude, I do respect the fact that the consumers of the web application platforms I develop are frequently utilizing what we would consider both legacy technologies (Internet Explorer 11) as well as modern browsers, and frequently am required to balance technological decisions in order to deliver the best user experience possible for users of all technologies.  I do not compromise my decision making to completely disregard users of legacy technologies and ensure that all decisions I make are adequately supported for all users, because I do not believe in offering a different internet experience to users of bleeding edge technologies and legacy technologies.  Yes, this means that bleeding edge features in browsers are sometimes not available to me and that I must find alternative implementations of those features, but this is a cost I would rather incur than make the users of the software that I create suffer.

I now await your apology for your public slander.

**Comment by ReLater** [comment]
> So what control the decision strategy to prioritize what in core or not?

Even if it's said already several times with other words above:

At least (for Joomla 3):
https://docs.joomla.org/Joomla_Browser_Support
https://downloads.joomla.org/technical-requirements

https://docs.joomla.org/Optional_Technical_Requirements

Further:
Volunteers that are willing to spend their time to develop solutions with these requirements in mind and that are willing to maintain features like that in the future.

That's the reason why a 3rd extension is better for a request like this because developers can define their own requirements, e.g. only browsers that support **modern and performant** feature-detection; in this case for attribute 'loading'. Thus any user can decide if he wants to have "modern" solutions or wants to be on the secure side with a pure 100% JavaScript solution or a solution for only 62% of possible visitors or ...

**Comment by infograf768** [comment]
Folks, as I use Mosaic since 1993, and my mail box gets filled with this Issue going nowhere (until further notice), I am deciding unilaterally to lock this Topic.

Thanks to all for giving your advice and knowledge.
